extern/usdhydra/addon/mx_nodes/nodes/base_node.py

```diff
@@ -0,0 +1,448 @@
+
+        for nd_input in mx_utils.get_nodedef_inputs(nodedef, True):
+            f = nd_input.getAttribute('uifolder')
+            if f and not getattr(self, self._folder_prop_name(f)):
+                continue
+
+            name = nd_input.getName()
+            if self.category in ("texture2d", "texture3d") and nd_input.getType() == 'filename':
+                split = layout.row(align=True).split(factor=0.4 if is_prop_area else 0.25, align=True)
+                col = split.column()
+                col.alignment='RIGHT' if is_prop_area else 'EXPAND'
+                col.label(text=nd_input.getAttribute('uiname') if nd_input.hasAttribute('uiname')
+                          else title_str(name))
+                col = split.column()
+                col.template_ID(self, self._input_prop_name(name),
+                                open="image.open", new="image.new")
+
+            else:
+                layout1 = layout
+                if is_prop_area:
+                    layout1 = layout1.split(factor=0.012, align=True)
+                    col = layout1.column()
+                    layout1 = layout1.column()
+                layout1.prop(self, self._input_prop_name(name))
+
+    def draw_node_view(self, context, layout):
+        from ...ui.material import USDHYDRA_MATERIAL_OP_invoke_popup_input_nodes
+        layout.use_property_split = True
+        layout.use_property_decorate = True
+        self.draw_buttons(context, layout)
+
+        for i, socket_in in enumerate(self.inputs):
+            nd = self.nodedef
+            uiname = mx_utils.get_attr(nd.getInput(socket_in.name), 'uiname',
+                                       title_str(nd.getInput(socket_in.name).getName()))
+            if socket_in.is_linked:
+                link = next((link for link in socket_in.links if link.is_valid), None)
+                if not link:
+                    continue
+
+                link = pass_node_reroute(link)
+                if not link or isinstance(link.from_node, bpy.types.NodeReroute):
+                    continue
+
+                split = layout.split(factor=0.4)
+                split_1 = split.split(factor=0.4)
+
+                row = split_1.row()
+                row.use_property_split = False
+                row.use_property_decorate = False
+                row.alignment = 'LEFT'
+                row.prop(socket_in, "show_expanded",
+                         icon="DISCLOSURE_TRI_DOWN" if socket_in.show_expanded else "DISCLOSURE_TRI_RIGHT",
+                         icon_only=True, emboss=False)
+                row = split_1.row()
+                row.alignment = 'RIGHT'
+                row.label(text=uiname)
+                row = split.row(align=True)
+                row.use_property_decorate = False
+
+                box = row.box()
+                box.scale_x = 0.7
+                box.scale_y = 0.5
+                box.emboss = 'NONE_OR_STATUS' if BLENDER_VERSION >= "3.0" else 'UI_EMBOSS_NONE_OR_STATUS'
```

Comment:
Are we going to support Blender versions lower than 3.0?
```suggestion
                box.emboss = 'NONE_OR_STATUS'
```